vignettes/rnoaa_openaq.Rmd

```diff
@@ -0,0 +1,156 @@
+  measurementsDelhi <- rbind(measurementsDelhi,
+                             aq_measurements(city = "Delhi",
+                                     parameter = "pm25",
+                                     date_from = "2016-03-01",
+                                 date_to = "2016-03-31",
+                                     limit = 1000, 
+                                     page = page)$results)
+}
+save(measurementsDelhi, file = "data/measurementsDelhi.RData")
+```
+
+We filter negative values.
+
+```{r, message = FALSE, warning = FALSE, eval = TRUE, echo = TRUE}
+library("dplyr")
+load("data/measurementsDelhi.RData")
+measurementsDelhi %>% head() %>% knitr::kable()
+measurementsDelhi <- filter(measurementsDelhi, value > 0)
+```
+
+We now transform these data into daily data.
+
+```{r, message = FALSE, warning = FALSE}
+# only keep stations with geographical information
+measurementsDelhi <- filter(measurementsDelhi, !is.na(latitude))
+# now transform to daily data
+measurementsDelhi <- measurementsDelhi %>% 
+  mutate(day = as.Date(dateLocal)) %>%
+  group_by(location, day) %>%
+  summarize(value = mean(value),
+            longitude = longitude[1],
+            latitude = latitude[1]) %>%
+  ungroup()
+measurementsDelhi %>% head() %>% knitr::kable()
+
+```
+
+Air quality and weather are correlated, so one could be interested in getting a time series of say temperature for the same location. The OpenAQ platform itself does not provide weather data but nearly all stations have geographical coordinates. Our goal here will be to use rnoaa to complement this table with precipitation and temperature.
+
+# Find weather stations
+
+For finding the right station(s) we shall use the `meteo_nearby_stations` function. It returns a list with the weather stations nearby each latitude/longitude given as arguments, respecting the other arguments such as maximal radius, first year with data, etc. For finding stations one might have to play a bit with the parameters until there is at least one station for each location.
+
+Here we query stations with a less than 15km distance from the air quality stations, with precipitation and temperature data, and with data starting from 2016. Note that the query takes a while.
+
+```{r, eval = FALSE, echo = TRUE}
+library("rnoaa")
+station_data <- ghcnd_stations()[[1]]
+lat_lon_df <- select(measurementsDelhi,
+                     location, 
+                     latitude,
+                     longitude) %>% unique() %>%
+  ungroup() %>%
+  rename(id = location) %>%
+  mutate(id = factor(id))
+
+stationsDelhi <- meteo_nearby_stations(lat_lon_df = as.data.frame(lat_lon_df),
+                                       station_data = station_data,
+                                       radius = 15,
+                                       year_min = 2016,
+                                       var = c("TAVG", "PRCP"))
+stationsDelhi <- unique(rbind_all(stationsDelhi) %>% select(- distance))
```

Comment:
@geanders switch to use `bind_rows` instead, as `rbind_all` is deprecated, and change in import statement in `rnoaa-package`